build/git.go

```diff
@@ -64,17 +64,19 @@ func getGitAttributes(ctx context.Context, contextPath string, dockerfilePath st
 		return res, nil
 	}
 
-	if sha, err := gitc.FullCommit(); err != nil {
-		return res, errors.Wrapf(err, "buildx: failed to get git commit")
-	} else if sha != "" {
-		if gitc.IsDirty() {
-			sha += "-dirty"
-		}
-		if setGitLabels {
-			res["label:"+specs.AnnotationRevision] = sha
-		}
-		if setGitInfo {
-			res["vcs:revision"] = sha
+	if gitc.HasCommit() {
```

Comment:
Could we do this as an error handler for `FullCommit` instead? 99.9% of builds will not have this special no-commit repo case and don't need the extra `git` binary call.